modules/shared/src/main/scala/org/tessellation/currency/dataApplication/package.scala

```diff
@@ -204,7 +320,18 @@ trait DataApplicationL0Service[F[_], D <: DataUpdate, DON <: DataOnChainState, D
 
 trait DataApplicationL1Service[F[_], D <: DataUpdate, DON <: DataOnChainState, DOF <: DataCalculatedState]
     extends DataApplicationService[F, D, DON, DOF]
-    with DataApplicationL1ContextualOps[F, D, DON, DOF]
+    with DataApplicationL1ContextualOps[F, D, DON, DOF] {
+
+  def requestDecoder(implicit f: Async[F]): EntityDecoder[F, DataTransactions] = {
+    implicit val dataUpdateDecoder: Decoder[Signed[D]] = Signed.decoder[D](dataDecoder)
```

Comment:
btw. Signed already has a decoder/encoder, so you just need to provide a decoder/encoder for D and for Signed you can take it implicitly from Signed companion object. There is no need to write it explicitly here as `Signed.decoder` or `Signed.encoder`

Reply:
Can you give me one example, I think I'm not understanding completely

Reply:
Instead of:
```scala
implicit val dataUpdateDecoder: Decoder[Signed[D]] = Signed.decoder[D](dataDecoder)
```
you can write just:
```
implicit val dataUpdateDecoder: Decoder[D] = dataDecoder
```
and you can decode `Signed[D]`, because there is already a decoder for `Signed[A]` so you just need to provide a decoder for `A`

Reply:
👍🏼